### PR TITLE
fix axislabel overlap test

### PR DESCRIPTION
**Pull request by KiaraEdTech**
Addresses the following issue: 
https://3.basecamp.com/3500782/buckets/1333707/todos/3860502110

The good in this solution is that it is only uses Vega expression and should only target date series in column charts (which is the issue about).
I figured out that the combination of no bound, zero flush and greedy overlap will solve the issue, although the first and last labels are reaching out of bound of axis, yet they still remain visible in the chart frame. 

The ideal solution could be aligning them to the respective axis edges,  but after trying out some methods I arrived to the conclusion that a more precise solution would take a disproportionate amount of time and energy. 

Testen on some charts on staging.
Fixture of the original issue chart included.

### CODE REVIEW

**Comment by KiaraEdTech** [comment]
Screenshot 2021-06-23 at 15 24 41
since I updated staging to test some charts, here is how the original issue item looked like before

**Comment by KiaraEdTech** [comment]
So this solution is going more directly to manipulation.
First, on mobile, we let Vega show labels with no overlap control. 
Then with similar conditions, we hide the ones between the first and the last in postprocessing, manipulating the DOM. There is a build in check to show all labels till 4 column, which seemed reasonable on a small screen. 

The weakness of this is that it does not check for dates only, so someone might want to do a 5 piece column chart with small labels, that won't cut it. It is not probable, but possible. In this edge case, the author could use bar charts.

An option would be a built in setting, where authors could select "do not hide labels automatically".

Tested on various cases with issue, did not tested yet on staging.

File: chartTypes/commonPostprocessings.js [hunk 1]
```diff
@@ -214,14 +216,44 @@ const addOutlineToAnnotationLabels = {
...
+};
+
+const showOnlyFirstAndLastLabel = {
+  process: function (svg, spec, item, toolRuntimeConfig) {
+    const svgDom = new JSDOM(svg);
+    const document = svgDom.window.document;
+
+    // this function only need to run if we are on mobile
+    if (spec.width > 400) return document.body.innerHTML;
+    console.log(item);
```
Comment (fromdusttilldawn) [inline chartTypes/commonPostprocessings.js:229]: Plz remove :)

Comment (KiaraEdTech) [reply chartTypes/commonPostprocessings.js]: thank you :)

**Review by fromdusttilldawn — APPROVED** [review]

**Comment by KiaraEdTech** [comment]
I scraped the database for edge cases and found a few with no dates that suffered from the hidden labels. I put a check to apply the manipulation only if the data has dates. 
A few examples on staging: 
https://q.st-staging.nzz.ch/item/f9b138787aed0dec49f73ab6174dc180
https://q.st-staging.nzz.ch/item/f149502e99d86e6bdb02e543f31fd2f7
I looked at the prod database, so there is a lot that only available on prod.

**Pull request merged**